airflow/models/dag.py

```diff
@@ -1840,6 +1840,7 @@ def set_task_instance_state(
         past: bool = False,
         commit: bool = True,
         session=NEW_SESSION,
+        group_id: str | None = None,
```

Comment:
I'm not sure I like the pattern of `set_task_instance_state` acting on a task instance, or a group when you pass this optional group_id (thus, ignoring the task_id completely). I think I'd rather see a `set_task_group_state` instead.

Reply:
I have implemented a new method `set_task_group_state` for this functionality. This looks much cleaner now.